common/types.go

```diff
@@ -284,12 +284,85 @@ type WriteResult struct {
 	Data map[string]any `json:"data,omitempty"` // optional
 }
 
-// DeleteResult is what's returned from deleting data via the Delete call.
+// DeleteResult represents the outcome of a single record delete operation.
 type DeleteResult struct {
 	// Success is true if deletion succeeded.
 	Success bool `json:"success"`
 }
 
+// BatchStatus describes the aggregate outcome of a batch operation.
+type BatchStatus string
+
+const (
+	BatchStatusSuccess BatchStatus = "success"
+	BatchStatusFailure BatchStatus = "failure"
+	BatchStatusPartial BatchStatus = "partial"
+)
+
+// BatchWriteType specifies the intended operation type within a batch modification.
+type BatchWriteType string
+
+const (
+	BatchWriteTypeCreate BatchWriteType = "create"
+	BatchWriteTypeUpdate BatchWriteType = "update"
+	BatchWriteTypeUpsert BatchWriteType = "upsert"
```

Comment:
no upsert yet, that's post MVP.  Supporting upsert will require more fields to be added to BatchWriteParam, which we don't want to do yet. That's next up, but we need to deliver create/update soon so that's why we're keep the MVP scope small.